src/main/java/com/ldtteam/structurize/util/IcuTranslatableContents.java

```diff
@@ -0,0 +1,169 @@
+package com.ldtteam.structurize.util;
+
+import com.ibm.icu.text.MessageFormat;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.datafixers.util.Either;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.locale.Language;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.ComponentContents;
+import net.minecraft.network.chat.ComponentUtils;
+import net.minecraft.network.chat.FormattedText;
+import net.minecraft.network.chat.FormattedText.ContentConsumer;
+import net.minecraft.network.chat.FormattedText.StyledContentConsumer;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.contents.TranslatableContents;
+import net.minecraft.world.entity.Entity;
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * Inspired by {@link TranslatableContents} to support things from {@link MessageFormat} (plurals, pronouns, etc.)
+ */
+public class IcuTranslatableContents implements ComponentContents
```

Comment:
Would probably put that in a separate PR

Reply:
can do later, but meh

Reply:
Done